File: decompiled/sources/com/hjq/base/action/ClickAction.java
package com.hjq.base.action;

import android.view.View;

public interface ClickAction extends View.OnClickListener {
    <V extends View> V findViewById(int i);

    void onClick(View view);

    void setOnClickListener(View.OnClickListener onClickListener, int... iArr);

    void setOnClickListener(View.OnClickListener onClickListener, View... viewArr);

    void setOnClickListener(int... iArr);

    void setOnClickListener(View... viewArr);

    /* renamed from: com.hjq.base.action.ClickAction$-CC  reason: invalid class name */
    public final /* synthetic */ class CC {
        public static void $default$onClick(ClickAction clickAction, View view) {
        }

        public static void $default$setOnClickListener(ClickAction _this, int... iArr) {
            _this.setOnClickListener((View.OnClickListener) _this, iArr);
        }

        public static void $default$setOnClickListener(ClickAction _this, View.OnClickListener onClickListener, int... iArr) {
            for (int findViewById : iArr) {
                _this.findViewById(findViewById).setOnClickListener(onClickListener);
            }
        }

        public static void $default$setOnClickListener(ClickAction _this, View... viewArr) {
            _this.setOnClickListener((View.OnClickListener) _this, viewArr);
        }

        public static void $default$setOnClickListener(ClickAction _this, View.OnClickListener onClickListener, View... viewArr) {
            for (View onClickListener2 : viewArr) {
                onClickListener2.setOnClickListener(onClickListener);
            }
        }
    }
}

File: decompiled/sources/com/hjq/base/action/ClickAction.kt
package com.hjq.base.action

import android.view.View

interface ClickAction : View.OnClickListener {
    fun <V : View> findViewById(id: Int): V

    override fun onClick(view: View) {
        // no-op
    }

    fun setOnClickListener(vararg viewIds: Int) {
        setOnClickListener(this, *viewIds)
    }

    fun setOnClickListener(onClickListener: View.OnClickListener, vararg viewIds: Int) {
        for (viewId in viewIds) {
            findViewById<View>(viewId).setOnClickListener(onClickListener)
        }
    }

    fun setOnClickListener(vararg views: View) {
        setOnClickListener(this, *views)
    }

    fun setOnClickListener(onClickListener: View.OnClickListener, vararg views: View) {
        for (view in views) {
            view.setOnClickListener(onClickListener)
        }
    }

    object CC {
        @JvmStatic
        fun `$default$onClick`(clickAction: ClickAction, view: View) {
            // no-op
        }

        @JvmStatic
        fun `$default$setOnClickListener`(clickAction: ClickAction, vararg viewIds: Int) {
            clickAction.setOnClickListener(clickAction, *viewIds)
        }

        @JvmStatic
        fun `$default$setOnClickListener`(
            clickAction: ClickAction,
            onClickListener: View.OnClickListener,
            vararg viewIds: Int
        ) {
            for (viewId in viewIds) {
                clickAction.findViewById<View>(viewId).setOnClickListener(onClickListener)
            }
        }

        @JvmStatic
        fun `$default$setOnClickListener`(clickAction: ClickAction, vararg views: View) {
            clickAction.setOnClickListener(clickAction, *views)
        }

        @JvmStatic
        fun `$default$setOnClickListener`(
            clickAction: ClickAction,
            onClickListener: View.OnClickListener,
            vararg views: View
        ) {
            for (view in views) {
                view.setOnClickListener(onClickListener)
            }
        }
    }
}
